### PR TITLE
fix some deprecations and other build warnings

### PR DESCRIPTION
**Pull request by raveit65**
I think it is save to drop the files which are only use by gnome-main-menu (EOL and no gtk+3 support).
If you think document-tile.c and directory-tile.c is use by other applications i can drop this commit.
I don't think that it is used by mate-control-center window itself, btw. gtk3-inspectector didn't show me the deprecated widgets from those files.
Please test.

### CODE REVIEW

**Comment by raveit65** [comment]
I just edited the GtkSeat ports and use GDK_SEAT_CAPABILITY_ALL here.
Please test again.

**Comment by raveit65** [comment]
@monsta 
Do you think **libslab : drop uneeded support for gnome-main-menu** is save to use?
I don't think that any other application use this code and i am really in doubt that @stefano-k 
will ever port this to gtk+-3.

**Comment by monsta** [comment]
I don't know any apps that use that code, and gnome-main-menu [looks dead](https://git.gnome.org/browse/gnome-main-menu/log). I was actually going to drop that code myself someday.

File: libslab/libslab-utils.c [hunk 1]
```diff
@@ -221,7 +221,11 @@ libslab_mate_desktop_item_open_help (MateDesktopItem *item)
 	if (doc_path) {
 		help_uri = g_strdup_printf ("help:%s", doc_path);
 
+#if GTK_CHECK_VERSION (3, 22, 0)
```
Comment (monsta) [inline libslab/libslab-utils.c]: I don't see `libslab_mate_desktop_item_open_help` being used anywhere, so it probably can be dropped.

Comment (raveit65) [inline libslab/libslab-utils.c]: You're right, a *grep* on my MATE folder gives me no results.
I will remove it.

**Comment by raveit65** [comment]
Done!
@monsta 
I will squash some commits after your approval.

**Comment by monsta** [comment]
Yes, sure, my additions can be squashed :slightly_smiling_face:

**Comment by raveit65** [comment]
All is rebased and some commits are squashed.
@monsta 
Anything else left or is this ready to go?

**Comment by monsta** [comment]
Commit with system-tile drop can be squashed too, I think it was only used in g-m-m

**Comment by raveit65** [comment]
@monsta Done!

Some more concerns ?   :-)

**Comment by monsta** [comment]
Looks ok, keybindings editor and m-c-c shell seem to work the same as before.

**Pull request merged**